SPIKE.md

```diff
@@ -0,0 +1,92 @@
+# Spike: Robust handling of ObjectID type for MongoDB
+
+Ref: https://github.com/strongloop/loopback-next/issues/3456
+
+## Table of contents
+
+- [The problem](#the-problem)
+- [Proposed solution](#proposed-solution)
+- [Approaches](#approaches)
+- [Developer Notes](#developer-notes)
+- [Implementation Proposal](#implementation)
+- [Follow-up task](#follow-up-task)
+
+## The Problem
+
+`ObjectID` is a native MongoDB datatype, which is used as the datatype of its primary key (_id_) field; other fields
+can also be of `ObjectID` type.
+
+In the current version of Juggler, the interpretation of `ObjectID` field is a sub-optimal experience supported by a complex
+determination process in an already complex codebase.
+
+Examples of bad experiences: 
+
+1. Fields need not be set to `mongodb: {dataType: 'ObjectID'}` to be interpreted as `ObjectID`.
```

Comment:
I find this sentence a bit difficult to understand. Are you trying to say this? 👇🏻 

```suggestion
1. Model properties must be defined as `{type: 'string', mongodb: {dataType: 'ObjectID'}}` to be interpreted as `ObjectID`.
```

Reply:
Nah, I am talking about the current behavior of interpreting any `ObjectID`-like string as an `ObjectID`.